clang/include/clang/Basic/SourceManager.h

```diff
@@ -1901,9 +1903,8 @@ class SourceManager : public RefCountedBase<SourceManager> {
 
   FileID getFileID(SourceLocation::UIntTy SLocOffset) const {
     // If our one-entry cache covers this offset, just return it.
-    if (isOffsetInFileID(LastFileIDLookup, SLocOffset))
+    if (SLocOffset >= LastLookupStartOffset && SLocOffset < LastLookupEndOffset)
```

Comment:
I'd somewhat prefer that `isOffsetInFileID` be taught this trick instead of here.  WDYT?

Reply:
good idea!

Reply:
hmm, moving to `isOffsetInFileID` makes the performance worse... https://llvm-compile-time-tracker.com/compare.php?from=50c6fcb2a1167b65255d2edce9ef34789b85a7a5&to=1ad5668b6f8e0a54fa258144a765ed3ea0ea8ad0&stat=instructions:u

Reply:
I’ve reverted the changes to `isOffsetInFileID` in this patch to maximize the overall performance gain.

I have some ideas for improving `isOffsetInFileID` and might explore them later -- though it’s not clear whether `isOffsetInFileID` will still be a hot method after this change.